src/KeyManager.ts

```diff
@@ -1,13 +1,454 @@
-import kbpgp from 'kbpgp'
+import fs from 'fs'
+import crypto from 'crypto'
+import chalk from 'chalk'
+import Path from 'path'
+import { promisify } from 'util'
 
-// we get back an object
-// not a series of modules I guess
-// otherwise the default is usually has to be import { default as ... } from ...
-// kbpgp.const.openpgp
-// so that's good
+
+
+// js imports
+const kbpgp = require('kbpgp')
+var F = kbpgp["const"].openpgp
+const zxcvbn = require('zxcvbn')
+
+// TODO: remember that the symmetric keys will just be stored in vault directories.
+// I feel the symmetric key funciton here are not needed.
+// This should be dealing with asymm key management
+// Should the ciphering happen through cryptor?
+  // I think it should
+
+// Make a base class (or interaface) 'Key'. It has most of the methods you see here. The symetric and asymmetric keys
+// can both have their implementations of it. For example, they'd both have different generateKeyPair
+// Or maybe just return two different objects from within the class, one for symmetric and one for asymmetric
+
+// We want a consistent and uniform library. So even if we change crypto libraries for symm or asymm, not much
+// changes for compositional code. KeyManager, returns a 'Key' instance. There can be difference classes for both
+// symm and asymm. Cryptor can then take these Key classes, which expose the crytpo functions. So even if the
+// crypto library changes, Cryptor doesn't have to cahnge and neither does polykey.
+
+
+type KeyPair = {
+  private: string,
+  public: string,
+  passphrase: string
+}
 
 class KeyManager {
+  // TODO: wouldn't keymanager have many sym keys keys to look after?
+  private keyPair: KeyPair = {private: '', public: '', passphrase: ''}
+  private identity: Object | undefined = undefined
+  private derivedKeys: Map<string, Buffer>
+  private passphrase!: string
+  private fs: typeof fs
+
+  storePath: string
+
+  constructor(
+    polyKeyPath: string = '~/.polykey/'
+  ) {
+    this.storePath = polyKeyPath
+    // Import keypair if it exists
+
+    this.derivedKeys = new Map()
+  }
+
+  // return {private: string, public: string}
+  async generateKeyPair(name: string, email: string, passphrase: string, numBits: number = 4096): Promise<KeyPair> {
+    // Validate passphrase
+    const passValidation = zxcvbn(passphrase)
+    // The following is an arbitrary delineation of desirable scores
+    if (passValidation.score < 2) {
+      console.log(chalk.red(`passphrase score for new keypair is below 2!`))
+    } else if (passValidation.score <4) {
+      console.log(chalk.yellow(`passphrase score for new keypair is below 4!`))
+    } else {
+      console.log(chalk.green(`passphrase score for new keypair is 4 or above.`))
```

Comment:
Hey you still using `chalk` here!

Reply:
oops! good spot, I've taken it out. This would be good to feed into our logger, but that is specific to our CLI. I think I can just remove the notifying of a good passphrase and make a bad passphrase throw an error.